src/components/todoList.tsx

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable jsx-a11y/control-has-associated-label */
+
+import { FC } from 'react';
+import { Todo as TodoTypes } from '../types/Todo';
+import React from 'react';
+import { Todo } from './Todo';
+import { TempTodo } from './TempTodo';
+
+interface Props {
+  onDelete: (id: number) => Promise<void>;
+  onEdit: (id: number, data: Partial<TodoTypes>) => void;
+  todos: TodoTypes[];
+  tempTodoTitle: string | null;
+}
+
+export const TodoList: FC<Props> = ({
+  todos,
+  tempTodoTitle,
+  onDelete,
+  onEdit,
+}) => {
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {todos.map(todo => (
+        <Todo key={todo.id} todo={todo} onDelete={onDelete} onEdit={onEdit} />
+      ))}
+
+      {tempTodoTitle && <TempTodo title={tempTodoTitle} />}
```

Comment:
it can be the same Todo component

for that you need to save the tempTodo instead of tempTodoTitle